cm-server/src/main/java/ro/msg/cm/controller/ImportDatabase.java

```diff
@@ -0,0 +1,111 @@
+package ro.msg.cm.controller;
+
+/**
+ * Created by oana on 5/16/17.
+ */
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.cm.model.Education;
+import ro.msg.cm.model.Tag;
+import ro.msg.cm.repository.EducationRepository;
+import ro.msg.cm.repository.TagRepository;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/import")
+public class ImportDatabase {
+
+    private final TagRepository tagRepository;
+    private final EducationRepository educationRepository;
+
+    @Autowired
+    public ImportDatabase(TagRepository tagRepository, EducationRepository educationRepository) {
+
+        this.tagRepository = tagRepository;
+        this.educationRepository = educationRepository;
+    }
+
+    @RequestMapping(value = "/education", method = RequestMethod.POST)
+    public void importEducation(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        printRequestDetail(false, request);
+        importCSV(Education.class, request.getInputStream(), this.educationRepository);
+    }
+
+    @RequestMapping(value = "/tag", method = RequestMethod.POST)
+    public void importTag(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        printRequestDetail(false, request);
+        importCSV(Tag.class, request.getInputStream(), this.tagRepository);
+    }
+
+    private void printRequestDetail(boolean debug, HttpServletRequest request) throws IOException {
+        if (!debug) return;
+        System.out.print(request.getMethod());
+        Enumeration params = request.getParameterNames();
+        while (params.hasMoreElements()) {
+            String paramName = (String) params.nextElement();
+            System.out.println("Parameter Name - " + paramName + ", Value - " + request.getParameter(paramName));
+        }
+        Enumeration headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
+            String headerName = (String) headerNames.nextElement();
+            System.out.println("Header Name - " + headerName + ", Value - " + request.getHeader(headerName));
+        }
+        String result = new BufferedReader(new InputStreamReader(request.getInputStream()))
+                .lines().collect(Collectors.joining("\n"));
+        System.out.println(result.length());
+        System.out.println(result);
+
+    }
+
+    private void importCSV(Class table, InputStream csvContent, CrudRepository rep) throws IOException {
```

Comment:
I propose that we use a library to do the "heavy lifting" CSV parsing for us. E.g. see [this S.O. question](https://stackoverflow.com/a/33912983/7612556). If not, maybe split this method into a couple of smaller methods.